core/trino-main/src/main/java/io/trino/operator/scalar/DateTimeFunctions.java

```diff
@@ -242,22 +242,48 @@ public static Slice toISO8601FromDate(@SqlType(StandardTypes.DATE) long date)
         return utf8Slice(formatter.print(DAYS.toMillis(date)));
     }
 
-    @ScalarFunction("from_iso8601_timestamp")
-    @LiteralParameters("x")
-    @SqlType("timestamp(3) with time zone")
-    public static long fromISO8601Timestamp(ConnectorSession session, @SqlType("varchar(x)") Slice iso8601DateTime)
-    {
-        DateTimeFormatter formatter = ISODateTimeFormat.dateTimeParser()
-                .withChronology(getChronology(session.getTimeZoneKey()))
-                .withOffsetParsed();
+@ScalarFunction("from_iso8601_timestamp")
```

Comment:
> fix issue trinodb#20515 add-iso8601-basic-format-support

https://trino.io/development/process#pull-request-and-commit-guidelines- is a commit message guideline in this project. Please rename to `Add support for basic format in from_iso8601_timestamp function` or something.